src/java.desktop/share/native/libawt/java2d/loops/Blit.c

```diff
@@ -74,19 +74,16 @@ Java_sun_java2d_loops_Blit_Blit
 
     srcInfo.bounds.x1 = srcx;
     srcInfo.bounds.y1 = srcy;
-    if (UNSAFE_TO_ADD(srcx, width) ||
-        UNSAFE_TO_ADD(srcy, height) ||
-        UNSAFE_TO_ADD(dstx, width) ||
-        UNSAFE_TO_ADD(dsty, height)) {
-        return;
-    }
-
-    srcInfo.bounds.x2 = srcx + width;
-    srcInfo.bounds.y2 = srcy + height;
+    srcInfo.bounds.x2 = UNSAFE_TO_ADD(srcx, width)
+                        ? clipInfo.bounds.x2 : (srcx + width);
+    srcInfo.bounds.y2 = UNSAFE_TO_ADD(srcy, height)
+                        ? clipInfo.bounds.y2 : (srcy + height);
     dstInfo.bounds.x1 = dstx;
     dstInfo.bounds.y1 = dsty;
-    dstInfo.bounds.x2 = dstx + width;
-    dstInfo.bounds.y2 = dsty + height;
+    dstInfo.bounds.x2 = UNSAFE_TO_ADD(dstx, width)
+                        ? clipInfo.bounds.x2 : (dstx + width);
+    dstInfo.bounds.y2 = UNSAFE_TO_ADD(dsty, height)
```

Comment:
why wouldn't you always want to limit it to the clip ?
I mean shouldn't it be like this ?
dstInfo.bounds.x2 = UNSAFE_TO_ADD(dstx, width)  ? clipInfo.bounds.x2 : (((dstx + width) > clipInfo.bounds.x2) ? clipInfo.bounds.x2 : (dstx + width));
or maybe a bit more readable as 
dstInfo.bounds.x2 = ((UNSAFE_TO_ADD(dstx, width) || ((dstx + width) > clipInfo.bounds.x2)) ? clipInfo.bounds.x2 : (dstx + width);

Reply:
That is because down below it anyway calls
`SurfaceData_IntersectBounds(&dstInfo.bounds, &clipInfo.bounds); `
so it should clip to clipInfo.bounds there I presume..

Reply:
@prrace Since it is going to clip dstInfo.bounds to clipInfo.bounds in SurfaceData_IntersectBounds, I believe is not necessary to do the duplicate clip here.
Let me know if you think otherwise..

Reply:
ok

Reply:
>That is because down below it anyway calls SurfaceData_IntersectBounds(&dstInfo.bounds, &clipInfo.bounds); 
so it should clip to clipInfo.bounds there.

In addition to intersecting dst with clip, you're also intersecting src with clip, which seems incorrect. A better approach might be to compute the drawable width based on the non-overflowing range for both src and dst, and then proceed with the original logic.

BTW I have not checked MaskBlit_MaskBlit yet it might require a similar update.

Reply:
We have not found any problem with that part of code so far..We have addressed the issue that was raised.
MaskBlit might require similar update but since it was not proven to be problematic (we waited for your feedback) and since RDP1 is approaching, we will integrate this and you can come back with POC for us to check..

Reply:
>We have not found any problem with that part of code so far..

This is incorrect because the clip bounds are relative to the destination (dst), so you cannot intersect them directly with the source (src) surface. The usage of UNSAFE_TO_SUB below is also wrong, we cannot just exists.

filed https://bugs.openjdk.org/browse/JDK-8358103